### ISSUE TITLE
Visual Studio errors and warnings

### ISSUE THREAD

**Issue by dzenanz** [InsightSoftwareConsortium/ITK#1431]
With the latest master (75ec4e508e31ef93cac17235db5a5600590bc1a8), VS2019 gives the below errors and warnings, most important of which are VNL link errors of the style `unresolved external symbol "public: __cdecl vnl_matrix<double>::vnl_matrix<double>`.
```text
In project ITKDCMTK_ExtProject (ExternalProjectTargets\ITKDCMTK_ExtProject\ITKDCMTK_ExtProject), file C:\Dev\ITK-2019\Modules\ThirdParty\DCMTK\CUSTOMBUILD, line 1: Warning : PNG support will be disabled because libpng directory was not specified. Correct path and re-enable DCMTK_WITH_PNG.
In project ITKDCMTK_ExtProject (ExternalProjectTargets\ITKDCMTK_ExtProject\ITKDCMTK_ExtProject), file C:\Dev\ITK-2019\Modules\ThirdParty\DCMTK\CUSTOMBUILD, line 1: Warning : TIFF support will be disabled because libtiff directory was not specified. Correct path and re-enable DCMTK_WITH_TIFF.
In project ITKDCMTK_ExtProject (ExternalProjectTargets\ITKDCMTK_ExtProject\ITKDCMTK_ExtProject), file C:\Dev\ITK-2019\Modules\ThirdParty\DCMTK\CUSTOMBUILD, line 1: Warning : ZLIB support will be disabled because zlib directory was not specified. Correct path and re-enable DCMTK_WITH_ZLIB.
In project ITKDCMTK_ExtProject (ExternalProjectTargets\ITKDCMTK_ExtProject\ITKDCMTK_ExtProject), file C:\Dev\ITK-2019\Modules\ThirdParty\DCMTK\CUSTOMBUILD, line 1: Warning : 
In project ITKGPUCommon, file C:\Dev\ITK-git\Modules\Core\GPUCommon\src\itkGPUContextManager.cxx, line 71: Warning C4996: 'clCreateCommandQueue': was declared deprecated
In project ITKGPUCommon, file C:\Program Files (x86)\OpenCL_SDK_Light\include\CL\cl.h, line 1359: Message : see declaration of 'clCreateCommandQueue'
In project ITKIODCMTK, file C:\Dev\ITK-2019\Modules\ThirdParty\DCMTK\ITKDCMTK_ExtProject-build\config\include\dcmtk\config\arith.h, line 1: Warning C4335: Mac file format detected: please convert the source file to either DOS or UNIX format
In project ITKIODCMTK, file C:\Dev\ITK-2019\Modules\ThirdParty\DCMTK\ITKDCMTK_ExtProject-build\config\include\dcmtk\config\arith.h, line 1: Warning C4335: Mac file format detected: please convert the source file to either DOS or UNIX format
In project ITKIODCMTK, file C:\Dev\ITK-2019\Modules\ThirdParty\DCMTK\ITKDCMTK_ExtProject-build\config\include\dcmtk\config\arith.h, line 1: Warning C4335: Mac file format detected: please convert the source file to either DOS or UNIX format
In project IOTransformDCMTK, file C:\Dev\ITK-2019\Modules\ThirdParty\DCMTK\ITKDCMTK_ExtProject-build\config\include\dcmtk\config\arith.h, line 1: Warning C4335: Mac file format detected: please convert the source file to either DOS or UNIX format
In project IOTransformDCMTKHeaderTest1, file C:\Dev\ITK-2019\Modules\ThirdParty\DCMTK\ITKDCMTK_ExtProject-build\config\include\dcmtk\config\arith.h, line 1: Warning C4335: Mac file format detected: please convert the source file to either DOS or UNIX format
In project ITKCommon1TestDriver, file C:\Dev\ITK-2019\Modules\Core\Common\test\itkPointGeometryTest.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_vector_ref<double>::vnl_vector_ref<double>(unsigned int,double *)" (??0?$vnl_vector_ref@N@@QEAA@IPEAN@Z) referenced in function "public: class vnl_vector_ref<double> __cdecl itk::Point<double,3>::GetVnlVector(void)" (?GetVnlVector@?$Point@N$02@itk@@QEAA?AV?$vnl_vector_ref@N@@XZ)
In project ITKCommon1TestDriver, file C:\Dev\ITK-2019\Modules\Core\Common\test\itkPointGeometryTest.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_vector_ref<double>::~vnl_vector_ref<double>(void)" (??1?$vnl_vector_ref@N@@QEAA@XZ) referenced in function "int __cdecl itkPointGeometryTest(int,char * * const)" (?itkPointGeometryTest@@YAHHQEAPEAD@Z)
In project ITKCommon1TestDriver, file C:\Dev\ITK-2019\bin\Debug\ITKCommon1TestDriver.exe, line 1: Error LNK1120: 2 unresolved externals
In project ITKGPUAnisotropicSmoothingTestDriver, file C:\Dev\ITK-2019\Modules\Filtering\GPUAnisotropicSmoothing\test\itkGPUGradientAnisotropicDiffusionImageFilterTest.obj, line 1: Error LNK2019: unresolved external symbol "public: static char const * __cdecl itk::GPUDenseFiniteDifferenceImageFilterKernel::GetOpenCLSource(void)" (?GetOpenCLSource@GPUDenseFiniteDifferenceImageFilterKernel@itk@@SAPEBDXZ) referenced in function "public: static char const * __cdecl itk::GPUDenseFiniteDifferenceImageFilter<class itk::GPUImage<float,2>,class itk::GPUImage<float,2>,class itk::GradientAnisotropicDiffusionImageFilter<class itk::GPUImage<float,2>,class itk::GPUImage<float,2> > >::GetOpenCLSource(void)" (?GetOpenCLSource@?$GPUDenseFiniteDifferenceImageFilter@V?$GPUImage@M$01@itk@@V12@V?$GradientAnisotropicDiffusionImageFilter@V?$GPUImage@M$01@itk@@V12@@2@@itk@@SAPEBDXZ)
In project ITKGPUAnisotropicSmoothingTestDriver, file C:\Dev\ITK-2019\bin\Debug\ITKGPUAnisotropicSmoothingTestDriver.exe, line 1: Error LNK1120: 1 unresolved externals
In project ITKGPUPDEDeformableRegistrationTestDriver, file C:\Dev\ITK-2019\Modules\Registration\GPUPDEDeformable\test\itkGPUDemonsRegistrationFilterTest.obj, line 1: Error LNK2019: unresolved external symbol "public: static char const * __cdecl itk::GPUDenseFiniteDifferenceImageFilterKernel::GetOpenCLSource(void)" (?GetOpenCLSource@GPUDenseFiniteDifferenceImageFilterKernel@itk@@SAPEBDXZ) referenced in function "public: static char const * __cdecl itk::GPUDenseFiniteDifferenceImageFilter<class itk::GPUImage<class itk::Vector<float,2>,2>,class itk::GPUImage<class itk::Vector<float,2>,2>,class itk::DemonsRegistrationFilter<class itk::GPUImage<float,2>,class itk::GPUImage<float,2>,class itk::GPUImage<class itk::Vector<float,2>,2> > >::GetOpenCLSource(void)" (?GetOpenCLSource@?$GPUDenseFiniteDifferenceImageFilter@V?$GPUImage@V?$Vector@M$01@itk@@$01@itk@@V12@V?$DemonsRegistrationFilter@V?$GPUImage@M$01@itk@@V12@V?$GPUImage@V?$Vector@M$01@itk@@$01@2@@2@@itk@@SAPEBDXZ)
In project ITKGPUPDEDeformableRegistrationTestDriver, file C:\Dev\ITK-2019\Modules\Registration\GPUPDEDeformable\test\itkGPUDemonsRegistrationFilterTest2.obj, line 1: Error LNK2001: unresolved external symbol "public: static char const * __cdecl itk::GPUDenseFiniteDifferenceImageFilterKernel::GetOpenCLSource(void)" (?GetOpenCLSource@GPUDenseFiniteDifferenceImageFilterKernel@itk@@SAPEBDXZ)
In project ITKGPUPDEDeformableRegistrationTestDriver, file C:\Dev\ITK-2019\bin\Debug\ITKGPUPDEDeformableRegistrationTestDriver.exe, line 1: Error LNK1120: 1 unresolved externals
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_matrix<double>::vnl_matrix<double>(class vnl_matrix<double> const &,class vnl_matrix<double> const &,struct vnl_tag_mul)" (??0?$vnl_matrix@N@@QEAA@AEBV0@0Uvnl_tag_mul@@@Z) referenced in function "public: class vnl_matrix<double> __cdecl vnl_matrix<double>::operator*(class vnl_matrix<double> const &)const " (??D?$vnl_matrix@N@@QEBA?AV0@AEBV0@@Z)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_matrix<double>::~vnl_matrix<double>(void)" (??1?$vnl_matrix@N@@QEAA@XZ) referenced in function "int `public: __cdecl vnl_matrix_ref<double>::vnl_matrix_ref<double>(unsigned int,unsigned int,double *)'::`1'::dtor$0" (?dtor$0@?0???0?$vnl_matrix_ref@N@@QEAA@IIPEAN@Z@4HA)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_vector<double>::vnl_vector<double>(class vnl_vector<double> const &,double,struct vnl_tag_mul)" (??0?$vnl_vector@N@@QEAA@AEBV0@NUvnl_tag_mul@@@Z) referenced in function "public: class vnl_vector<double> __cdecl vnl_vector<double>::operator*(double)const " (??D?$vnl_vector@N@@QEBA?AV0@N@Z)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_vector<double>::vnl_vector<double>(class vnl_matrix<double> const &,class vnl_vector<double> const &,struct vnl_tag_mul)" (??0?$vnl_vector@N@@QEAA@AEBV?$vnl_matrix@N@@AEBV0@Uvnl_tag_mul@@@Z) referenced in function "class vnl_vector<double> __cdecl operator*<double>(class vnl_matrix<double> const &,class vnl_vector<double> const &)" (??$?DN@@YA?AV?$vnl_vector@N@@AEBV?$vnl_matrix@N@@AEBV0@@Z)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_vector<double>::~vnl_vector<double>(void)" (??1?$vnl_vector@N@@QEAA@XZ) referenced in function "public: __cdecl itk::Array<double>::~Array<double>(void)" (??1?$Array@N@itk@@QEAA@XZ)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_matrix<unsigned int>::~vnl_matrix<unsigned int>(void)" (??1?$vnl_matrix@I@@QEAA@XZ) referenced in function "public: __cdecl itk::Array2D<unsigned int>::~Array2D<unsigned int>(void)" (??1?$Array2D@I@itk@@QEAA@XZ)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_vector<unsigned int>::~vnl_vector<unsigned int>(void)" (??1?$vnl_vector@I@@QEAA@XZ) referenced in function "public: __cdecl itk::Array<unsigned int>::~Array<unsigned int>(void)" (??1?$Array@I@itk@@QEAA@XZ)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_vector<unsigned long>::~vnl_vector<unsigned long>(void)" (??1?$vnl_vector@K@@QEAA@XZ) referenced in function "public: __cdecl itk::Array<unsigned long>::~Array<unsigned long>(void)" (??1?$Array@K@itk@@QEAA@XZ)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_matrix<unsigned long>::~vnl_matrix<unsigned long>(void)" (??1?$vnl_matrix@K@@QEAA@XZ) referenced in function "public: __cdecl itk::Array2D<unsigned long>::~Array2D<unsigned long>(void)" (??1?$Array2D@K@itk@@QEAA@XZ)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_matrix<long>::~vnl_matrix<long>(void)" (??1?$vnl_matrix@J@@QEAA@XZ) referenced in function "public: void * __cdecl vnl_matrix<long>::`vector deleting destructor'(unsigned int)" (??_E?$vnl_matrix@J@@QEAAPEAXI@Z)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_vector_ref<double>::vnl_vector_ref<double>(unsigned int,double *)" (??0?$vnl_vector_ref@N@@QEAA@IPEAN@Z) referenced in function "public: class vnl_vector<double> __cdecl itk::Point<double,3>::GetVnlVector(void)const " (?GetVnlVector@?$Point@N$02@itk@@QEBA?AV?$vnl_vector@N@@XZ)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_vector_ref<double>::~vnl_vector_ref<double>(void)" (??1?$vnl_vector_ref@N@@QEAA@XZ) referenced in function "public: class vnl_vector<double> __cdecl itk::Point<double,3>::GetVnlVector(void)const " (?GetVnlVector@?$Point@N$02@itk@@QEBA?AV?$vnl_vector@N@@XZ)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\bin\Debug\ITKRegistrationCommonTestDriver.exe, line 1: Error LNK1120: 12 unresolved externals
In project ITKIODCMTKHeaderTest1, file C:\Dev\ITK-2019\Modules\ThirdParty\DCMTK\ITKDCMTK_ExtProject-build\config\include\dcmtk\config\arith.h, line 1: Warning C4335: Mac file format detected: please convert the source file to either DOS or UNIX format
```
This was first discussed in #1430.

**Comment by blowekamp** [comment]
@dzenanz We are scheduled for the v5.1b2 release very soon. Will you be able to look into this issue?

**Comment by dzenanz** [comment]
I guess I will have to 😄

**Comment by dzenanz** [comment]
Removing warnings and the errors fixed by #1436, this is an up-to-date list:
```text
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_matrix<double>::vnl_matrix<double>(class vnl_matrix<double> const &,class vnl_matrix<double> const &,struct vnl_tag_mul)" (??0?$vnl_matrix@N@@QEAA@AEBV0@0Uvnl_tag_mul@@@Z) referenced in function "public: class vnl_matrix<double> __cdecl vnl_matrix<double>::operator*(class vnl_matrix<double> const &)const " (??D?$vnl_matrix@N@@QEBA?AV0@AEBV0@@Z)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_matrix<double>::~vnl_matrix<double>(void)" (??1?$vnl_matrix@N@@QEAA@XZ) referenced in function "int `public: __cdecl vnl_matrix_ref<double>::vnl_matrix_ref<double>(unsigned int,unsigned int,double *)'::`1'::dtor$0" (?dtor$0@?0???0?$vnl_matrix_ref@N@@QEAA@IIPEAN@Z@4HA)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_vector<double>::vnl_vector<double>(class vnl_vector<double> const &,double,struct vnl_tag_mul)" (??0?$vnl_vector@N@@QEAA@AEBV0@NUvnl_tag_mul@@@Z) referenced in function "public: class vnl_vector<double> __cdecl vnl_vector<double>::operator*(double)const " (??D?$vnl_vector@N@@QEBA?AV0@N@Z)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_vector<double>::vnl_vector<double>(class vnl_matrix<double> const &,class vnl_vector<double> const &,struct vnl_tag_mul)" (??0?$vnl_vector@N@@QEAA@AEBV?$vnl_matrix@N@@AEBV0@Uvnl_tag_mul@@@Z) referenced in function "class vnl_vector<double> __cdecl operator*<double>(class vnl_matrix<double> const &,class vnl_vector<double> const &)" (??$?DN@@YA?AV?$vnl_vector@N@@AEBV?$vnl_matrix@N@@AEBV0@@Z)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_vector<double>::~vnl_vector<double>(void)" (??1?$vnl_vector@N@@QEAA@XZ) referenced in function "public: __cdecl itk::Array<double>::~Array<double>(void)" (??1?$Array@N@itk@@QEAA@XZ)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_matrix<unsigned int>::~vnl_matrix<unsigned int>(void)" (??1?$vnl_matrix@I@@QEAA@XZ) referenced in function "public: __cdecl itk::Array2D<unsigned int>::~Array2D<unsigned int>(void)" (??1?$Array2D@I@itk@@QEAA@XZ)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_vector<unsigned int>::~vnl_vector<unsigned int>(void)" (??1?$vnl_vector@I@@QEAA@XZ) referenced in function "public: __cdecl itk::Array<unsigned int>::~Array<unsigned int>(void)" (??1?$Array@I@itk@@QEAA@XZ)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_vector<unsigned long>::~vnl_vector<unsigned long>(void)" (??1?$vnl_vector@K@@QEAA@XZ) referenced in function "public: __cdecl itk::Array<unsigned long>::~Array<unsigned long>(void)" (??1?$Array@K@itk@@QEAA@XZ)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_matrix<unsigned long>::~vnl_matrix<unsigned long>(void)" (??1?$vnl_matrix@K@@QEAA@XZ) referenced in function "public: __cdecl itk::Array2D<unsigned long>::~Array2D<unsigned long>(void)" (??1?$Array2D@K@itk@@QEAA@XZ)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_matrix<long>::~vnl_matrix<long>(void)" (??1?$vnl_matrix@J@@QEAA@XZ) referenced in function "public: void * __cdecl vnl_matrix<long>::`vector deleting destructor'(unsigned int)" (??_E?$vnl_matrix@J@@QEAAPEAXI@Z)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_vector_ref<double>::vnl_vector_ref<double>(unsigned int,double *)" (??0?$vnl_vector_ref@N@@QEAA@IPEAN@Z) referenced in function "public: class vnl_vector<double> __cdecl itk::Point<double,3>::GetVnlVector(void)const " (?GetVnlVector@?$Point@N$02@itk@@QEBA?AV?$vnl_vector@N@@XZ)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\Modules\Registration\Common\test\itkMultiResolutionImageRegistrationMethodTest_1.obj, line 1: Error LNK2019: unresolved external symbol "public: __cdecl vnl_vector_ref<double>::~vnl_vector_ref<double>(void)" (??1?$vnl_vector_ref@N@@QEAA@XZ) referenced in function "public: class vnl_vector<double> __cdecl itk::Point<double,3>::GetVnlVector(void)const " (?GetVnlVector@?$Point@N$02@itk@@QEBA?AV?$vnl_vector@N@@XZ)
In project ITKRegistrationCommonTestDriver, file C:\Dev\ITK-2019\bin\Debug\ITKRegistrationCommonTestDriver.exe, line 1: Error LNK1120: 12 unresolved externals
In project ITKGPUAnisotropicSmoothingTestDriver, file C:\Dev\ITK-2019\Modules\Filtering\GPUAnisotropicSmoothing\test\itkGPUGradientAnisotropicDiffusionImageFilterTest.obj, line 1: Error LNK2019: unresolved external symbol "public: static char const * __cdecl itk::GPUDenseFiniteDifferenceImageFilterKernel::GetOpenCLSource(void)" (?GetOpenCLSource@GPUDenseFiniteDifferenceImageFilterKernel@itk@@SAPEBDXZ) referenced in function "public: static char const * __cdecl itk::GPUDenseFiniteDifferenceImageFilter<class itk::GPUImage<float,2>,class itk::GPUImage<float,2>,class itk::GradientAnisotropicDiffusionImageFilter<class itk::GPUImage<float,2>,class itk::GPUImage<float,2> > >::GetOpenCLSource(void)" (?GetOpenCLSource@?$GPUDenseFiniteDifferenceImageFilter@V?$GPUImage@M$01@itk@@V12@V?$GradientAnisotropicDiffusionImageFilter@V?$GPUImage@M$01@itk@@V12@@2@@itk@@SAPEBDXZ)
In project ITKGPUAnisotropicSmoothingTestDriver, file C:\Dev\ITK-2019\bin\Debug\ITKGPUAnisotropicSmoothingTestDriver.exe, line 1: Error LNK1120: 1 unresolved externals
In project ITKGPUPDEDeformableRegistrationTestDriver, file C:\Dev\ITK-2019\Modules\Registration\GPUPDEDeformable\test\itkGPUDemonsRegistrationFilterTest.obj, line 1: Error LNK2019: unresolved external symbol "public: static char const * __cdecl itk::GPUDenseFiniteDifferenceImageFilterKernel::GetOpenCLSource(void)" (?GetOpenCLSource@GPUDenseFiniteDifferenceImageFilterKernel@itk@@SAPEBDXZ) referenced in function "public: static char const * __cdecl itk::GPUDenseFiniteDifferenceImageFilter<class itk::GPUImage<class itk::Vector<float,2>,2>,class itk::GPUImage<class itk::Vector<float,2>,2>,class itk::DemonsRegistrationFilter<class itk::GPUImage<float,2>,class itk::GPUImage<float,2>,class itk::GPUImage<class itk::Vector<float,2>,2> > >::GetOpenCLSource(void)" (?GetOpenCLSource@?$GPUDenseFiniteDifferenceImageFilter@V?$GPUImage@V?$Vector@M$01@itk@@$01@itk@@V12@V?$DemonsRegistrationFilter@V?$GPUImage@M$01@itk@@V12@V?$GPUImage@V?$Vector@M$01@itk@@$01@2@@2@@itk@@SAPEBDXZ)
In project ITKGPUPDEDeformableRegistrationTestDriver, file C:\Dev\ITK-2019\Modules\Registration\GPUPDEDeformable\test\itkGPUDemonsRegistrationFilterTest2.obj, line 1: Error LNK2001: unresolved external symbol "public: static char const * __cdecl itk::GPUDenseFiniteDifferenceImageFilterKernel::GetOpenCLSource(void)" (?GetOpenCLSource@GPUDenseFiniteDifferenceImageFilterKernel@itk@@SAPEBDXZ)
In project ITKGPUPDEDeformableRegistrationTestDriver, file C:\Dev\ITK-2019\bin\Debug\ITKGPUPDEDeformableRegistrationTestDriver.exe, line 1: Error LNK1120: 1 unresolved externals
```

**Comment by dzenanz** [comment]
This happens with BUILD_SHARED_LIBS ON and full legacy remove:
ITK_FUTURE_LEGACY_REMOVE ON
ITK_LEGACY_REMOVE ON
VNL_CONFIG_LEGACY_METHODS OFF
VXL_LEGACY_FUTURE_REMOVE ON

**Comment by thewtex** [comment]
Has this been resolved by https://github.com/InsightSoftwareConsortium/ITK/pull/1436 ?

**Comment by dzenanz** [comment]
No, only two of these errors were resolved by #1436.

**Comment by dzenanz** [comment]
Resolved by https://github.com/vxl/vxl/pull/726. I guess that problem was incorrect rebuild behavior. A clean+build of the affected project resolves the issue.
